File: Q/python/code.py
from typing import List, Tuple


def is_on_one_line(points: List[Tuple[int]]) -> bool:
    # your code goes here
    return False

n = int(input())
points = []
for i in range(n):
    x, y = map(int, input().split())
    points.append((x, y))

if is_on_one_line(points):
    print('YES')
else:
    print('NO')

File: Q/js/code.js
class Point {
    constructor(x, y) {
        this.x = x;
        this.y = y;
    }
}

function isOnOneLine(points) {
    // your code goes here
    return false;
}

const _readline = require('readline');

const _reader = _readline.createInterface({
    input: process.stdin
});

const _inputLines = [];
let _curLine = 0;

_reader.on('line', line => {
    _inputLines.push(line);
});

process.stdin.on('end', solve);

function solve() {
    const n = readInt();
    const points = [];
    for (let i = 0; i < n; i++) {
        const coordinates = readArray();
        points.push(new Point(coordinates[0], coordinates[1]));
    }
    if (isOnOneLine(points)) {
        console.log("YES");
    } else {
        console.log("NO");
    }
}

function readInt() {
    const n = Number(_inputLines[_curLine]);
    _curLine++;
    return n;
}

function readArray() {
    var arr = _inputLines[_curLine].trim(" ").split(" ").map(num => Number(num));
    _curLine++;
    return arr;
}